circuit/primitives/src/event_signature.rs

```diff
@@ -170,4 +237,89 @@ pub mod tests {
             decode_signature(valid_signature_transfer_confirm_event.as_bytes().to_vec());
         assert_eq!(decode_res, Err("Signature must have non-empty event name"))
     }
+
+    #[test]
+    fn successfully_decodes_dfd_for_3_parallel_events() {
+        let valid_dfd = "A,B,C";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Ok(vec![vec![b"A".to_vec(), b"B".to_vec(), b"C".to_vec(),]])
+        )
+    }
+    #[test]
+    fn successfully_decodes_dfd_for_3_parallel_events_in_single_brackets() {
+        let valid_dfd = "(A,B,C)";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Ok(vec![vec![b"A".to_vec(), b"B".to_vec(), b"C".to_vec(),]])
+        )
+    }
+    #[test]
+    fn successfully_decodes_dfd_for_3_parallel_events_in_triple_brackets() {
+        let valid_dfd = "(((A,B,C)))";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Ok(vec![vec![b"A".to_vec(), b"B".to_vec(), b"C".to_vec(),]])
+        )
+    }
+    #[test]
+    fn fails_to_decode_dfd_for_3_parallel_events_with_incorrect_closing_brackets() {
+        let valid_dfd = "(A,B,C";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(decode_res, Err("DFD Decoder - too many opening brackets"))
+    }
+    #[test]
+    fn fails_to_decode_dfd_for_3_parallel_events_with_incorrect_opening_brackets() {
+        let valid_dfd = "A,B,C)";
+        // Important! If using .encode() instead of .as_bytes() + .to_vec(),
+        //  SCALE adds additional byte "92" to event name
+        let decode_res = decode_dfd(valid_dfd.as_bytes().to_vec());
+        assert_eq!(
+            decode_res,
+            Err("DFD Decoder - attempt to edit step at incorrect depth")
+        )
+    }
+
+    #[test]
+    fn successfully_decodes_dfd_for_3_sequential_events() {
+        let valid_dfd = "(A(B(C)))";
```

Comment:
can we test with some actual examples? Like transfer, swap etc... ?

Reply:
added a test case for 32b long values that are IDs